webui/src/components/editfiles.jsx

```diff
@@ -518,15 +516,31 @@ export const FileRecordRow = React.createClass({
         return {
             open: false,
             remove: false,
+            downloadsTot: 0,
         };
     },
 
+    componentDidMount() {
```

Comment:
I added the ajax request in componentDidMount here, and called then to update the downloadsTot in the state. If we don't use promises, should I change this to only a callback function?

Reply:
Yes, I'd replace it with:
````
serverCache.getFileStatistics(file.version_id, file.bucket, (mystat) => {
    let downloadsTot = mystat.buckets[0].value;
    this.setState({ downloadsTot });
})
````

And in serverCache:
````
getFileStatistics(file_id, bucket_id, successFn) {
    var data = {
        "mystat": {
            "stat": "bucket-file-download-total",
            "params": {
                "start_date": "2010-01-01",
                "bucket_id": bucket_id,
             }
         }
     };
     ajaxPost({
                 url: apiUrls.statistics(),
                 params: data,
                 successFn: response => { 
                     successFn(response.mystat);
                 },
             });
         });
     }
  };		  
};
````

Reply:
@emanueldima changed it to this, it works, thank you